util/stream_request.py

```diff
@@ -158,14 +158,14 @@ def calculate_derived_products(self):
         # Calculate all internal-only data products
         for sk in self.datasets:
             if not sk.is_virtual:
-                self.datasets[sk].calculate_all()
+                self.datasets[sk].calculate_all(ignore_missing_optional_params=False)
```

Comment:
is there a reason not to allow this for an internal data set? Consider the case where we may want to consolidate streams. A CTD on a glider may have a computed pressure value, whereas another CTD may have an algorithm to compute it. This feature would make it possible to use the same data product in the stream to handle either source.